### DISCUSSION TITLE
multiple sequential `clipboard.writeText`?

### DISCUSSION THREAD

**Discussion by ramiroaraujo** [johnlindquist/kit#1370] [Q&A]
I'm refactoring a clipboard split/merge script to use the Kit's clipboard history instead of Alfred's sqlite. When splitting an existing clipboard content by, say `\n`, I need to perform multiple `await clipboard.writeText()` with each line. This doesn't seem to work correctly, and for now the only hacky workaround is to add `await wait(150)` in between each write. I had to fish for that 150ms, since using a lower value, sometimes, missed an entry.

I'm running Kit `v1.99.47` in MacOS `14.0 (23A344)`
Here's a basic proof of concept:
```
import "@johnlindquist/kit"

const entries = ['one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine', 'ten']
for (const entry of entries) {
  await clipboard.writeText(entry)
  await wait(150) //comment this out and will only persist the last entry in the clipboard
}
```

**Comment by ramiroaraujo** [comment]
Probably a related issue:
If I `copy` really fast in the OS, some copied items are missed in the clipboard history. I can see it's copied correctly since Alfred's clipboard history shows them. I also tried quitting Alfred to check if it's interfering and I had the same results

**Reply by ramiroaraujo** [reply]
and BTW it's fast but not that fast, it's happening right now on normal multi-copy scenarios

**Reply by johnlindquist** [reply]
@ramiroaraujo Thanks, yeah, I do have a `debounce` on the clipboard change detection around ~1 second, I might be able to remove it or lower it drastically. I used to have a different approach that struggled with parsing large images, huge bodies of text, etc from the clipboard, but it's in a much better spot now. I'll stress test it and see what I can do.

**Reply by ramiroaraujo** [reply]
awesome!

**Reply by johnlindquist** [reply]
@ramiroaraujo Can you try this build, I moved the debounce down to 100ms

https://github.com/johnlindquist/kitapp/releases/tag/v1.99.76
